src/3d/qgslayoutitem3dmap.cpp

```diff
@@ -0,0 +1,67 @@
+/***************************************************************************
+  qgslayoutitem3dmap.cpp
+  --------------------------------------
+  Date                 : August 2018
+  Copyright            : (C) 2018 by Martin Dobias
+  Email                : wonder dot sk at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgslayoutitem3dmap.h"
+
+#include "qgs3dmapscene.h"
+#include "qgs3dutils.h"
+#include "qgscameracontroller.h"
+#include "qgslayout.h"
+#include "qgslayoutitemregistry.h"
+#include "qgsoffscreen3dengine.h"
+
+
+QgsLayoutItem3DMap::QgsLayoutItem3DMap( QgsLayout *layout )
+  : QgsLayoutItem( layout )
+{
+
+}
+
+QgsLayoutItem3DMap *QgsLayoutItem3DMap::create( QgsLayout *layout )
+{
+  return new QgsLayoutItem3DMap( layout );
+}
+
+int QgsLayoutItem3DMap::type() const
+{
+  return QgsLayoutItemRegistry::Layout3DMap;
+}
+
+void QgsLayoutItem3DMap::draw( QgsLayoutItemRenderContext &context )
+{
+  QgsOffscreen3DEngine engine;
+  QSizeF sizePixels = mLayout->renderContext().measurementConverter().convert( sizeWithUnits(), QgsUnitTypes::LayoutPixels ).toQSizeF();
+  engine.setSize( QSize( qCeil( sizePixels.width() ), qCeil( sizePixels.height() ) ) );
```

Comment:
Can you use std instead of the qt ceil method?

Reply:
Actually I started with std but then thought that ```qCeil(x)``` is more elegant than ```static_cast<int>( std::ceil(x))``` - but I can switch back...

Reply:
I'd switch back. I think those qt methods have a very short life expectancy - I'd be very surprised to see them survive into qt 6.